### DISCUSSION TITLE
Force overwrite on config file

### DISCUSSION THREAD

**Discussion by fjalcroft** [wixtoolset/issues#8120] [Questions]
Hi
New to WiX project installer, be gentle :)

Below is an example of the files I am including into my installer file (this is a Windows application)
  <Fragment>
    <ComponentGroup Id="ProductComponents" Directory="INSTALLFOLDER">
      <Component Id="Client" Guid="99dbab9e-0bc1-4db0-8160-6c8d76e99ae4">
        <File Id="Client" Name="Client.exe" Source="..\Client\bin\Release\"></File>
        <File Id="Client_Icon" Name="pv_orange.ico" Source="..\Client\bin\Release\"></File>
        <File Id="Client_appconfig" Name="Client.exe.config" Source="..\Client\App.Release.config"></File>
        <File Id="Client_CustomAppConfig" Name="Assured.CustomAppConfig.dll" Source="..\Client\bin\Release\"></File>
        <File Id="Client_MSTSCLib" Name="AxInterop.MSTSCLib.dll" Source="..\Client\bin\Release\"></File>
        <File Id="Client_Crypto" Name="Crypto.dll" Source="..\Client\bin\Release\"></File>
      </Component>
    </ComponentGroup>
  </Fragment>

I want to force overwrite of the app.config file. Our application updates the app.config, and if the file has been modified the installer will not replace it with that from the MSI file.

Can this be done? If so, how?

Many thanks.

James

**Comment by chrpai** [comment]
But then you have another problem... you'll lose the updates that your application put in the app.config.

Easiest solution exists outside of the installer.  Don't write to the app.config.  Instead write to an override.app.config and have your code use the original app.config as default values and override.app.config as override values.  This way date create / date modification stay the same and MSI always overwrites the app.config automatically.

In fact, you'll see the appSettings element already has a @file attribute to make this super easy.

https://learn.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/appsettings/appsettings-element-for-configuration

**Reply by fjalcroft** [reply]
Thanks chrpai
I understand what you are saying, but in this circumstance, I want to replace the app.config even if it has been changed. The contents of this config file are encrypted values to connect to web service and we have had to change the encryption algorithm.
From your suggestion, I guess we should change our app to write to an override file, with the app checking for changes and push these changes through to the override file. I can see how this can work. Need to get our dev team busy.
Thanks again
James

**Comment by robmen** [comment]
What if you make sure to reset the Created and Modified date/time on the config file are set the same after making changes? Then the Windows Installer should upgrade the file.

**Reply by chrpai** [reply]
Probably a.race condition there. He'd need to deploy a release that contained that logic before the next release would be able to leverage it.0
